src/turbopuffer.ts

```diff
@@ -39,12 +39,13 @@ export type FilterOperator =
   | "NotIGlob"
   | "And"
   | "Or";
+export type FilterFieldsOperator = "And" | "Or";
 export type FilterValue = Exclude<AttributeType, null>;
-export type FilterEntry = FilterOperator | FilterValue;
-export type FilterParam = FixedLengthArray<FilterEntry | FilterEntry[], 2>;
-export type Filters = {
-  [key: string]: FilterParam[];
+export type FilterParam = [FilterOperator, FilterParam | FilterValue];
```

Comment:
should be `Array<FilterParam> | FilterValue` (1st case for And/Or, 2nd case for everything else)